### PR TITLE
CaptureAppender and RequestLogEvent improvements

### PR DESCRIPTION
**Pull request by stevenschlansker**
in e.g. a test case, write:
```java
List<ILoggingEvent> logs = CaptureAppender.capture(MyClass.class);
```
and you get a live view of logs emitted by the class

### CODE REVIEW

File: core/src/main/java/com/opentable/logging/CommonLogHolder.java [hunk 1]
```diff
@@ -61,6 +61,23 @@
 
     private CommonLogHolder() { }
 
+    /**
+     * Mock out the environment.  You probably don't want to do this.
+     */
+    public static void setEnvironment(String otEnv, String otEnvType, String otEnvLocation, String otEnvFlavor) {
```
Comment (pennello) [inline core/src/main/java/com/opentable/logging/CommonLogHolder.java]: Can we limit the publicity of this method?

File: core/src/main/java/com/opentable/logging/CommonLogHolder.java [hunk 2]
```diff
@@ -61,6 +61,23 @@
...
+        OT_ENV = otEnv;
+        OT_ENV_TYPE = otEnvType;
+        OT_ENV_LOCATION = otEnvLocation;
+        OT_ENV_FLAVOR = otEnvFlavor;
+    }
+
+    /**
+     * Set main {@code OT_ENV} variables to {@code test}.
+     */
+    public static void forceTestEnvironment() {
```
Comment (pennello) [inline core/src/main/java/com/opentable/logging/CommonLogHolder.java]: This one too?

**Comment by pennello** [comment]
Travis failing:

    [INFO] Downloading: https://oss.sonatype.org/content/repositories/releases/com/opentable/logging/otj-otl/0.5.1/otj-otl-0.5.1.pom
    [INFO] Downloading: https://repository.apache.org/releases/com/opentable/logging/otj-otl/0.5.1/otj-otl-0.5.1.pom
    [WARNING] The POM for com.opentable.logging:otj-otl:jar:0.5.1 is missing, no dependency information available
    [INFO] Downloading: https://oss.sonatype.org/content/repositories/releases/com/opentable/logging/otj-otl/0.5.1/otj-otl-0.5.1.jar
    [INFO] Downloading: https://repository.apache.org/releases/com/opentable/logging/otj-otl/0.5.1/otj-otl-0.5.1.jar
    # ...
    [ERROR] Failed to execute goal on project otj-logging: Could not resolve dependencies for project com.opentable.components:otj-logging:jar:2.7.0-SNAPSHOT: Failure to find com.opentable.logging:otj-otl:jar:0.5.1 in http://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]

File: core/src/main/java/com/opentable/logging/CommonLogHolder.java (hunk 1 above)
Comment (stevenschlansker) [reply core/src/main/java/com/opentable/logging/CommonLogHolder.java]: Not without requiring test cases that need this to go in the `logging` package.

**Pull request merged**